src/helpers.js

```diff
@@ -71,29 +131,135 @@ const getPublisherRulesV1 = publishers => {
+    if (apply === 'in') {
+        return Object.fromEntries(Object.entries({ ...CountryTiers })
+            .filter(([key, value]) => {
+                return location.in.some(x => x === key || value.countries.includes(x))
+            })
+            .map(([key, value]) => {
+                if (location.in.includes(key)) {
+                    return [key, value]
+                }
+
+                const countries = [...value.countries].filter(x => location.in.includes(x))
+                return [key, { ...value, countries }]
+            })
+        )
+    } else if (apply === 'nin') {
+        return Object.fromEntries(Object.entries(CountryTiers)
+            .filter(([key, value]) => {
+                return !location.nin.includes(key) || !value.countries.every(x => location.nin.includes(x))
+            })
+            .map(([key, value]) => {
+                const countries = [...value.countries].filter(x => !location.nin.includes(x))
+                return [key, { ...value, countries }]
+            }))
+    }
+
+    return { ...CountryTiers }
+}
+
+const getPriceRulesV1 = ({ audienceInput, countryTiersCoefficients, pricingBounds, decimals }) => {
+    const { inputs } = audienceInput
+    const { location = {}, categories = {}, publishers = {}, advanced = {} } = inputs
+    // using floats for easier math as it is no crucial
+    const userPricingBounds = {
+        min: parseFloat(formatUnits(pricingBounds.min, decimals)),
+        max: parseFloat(formatUnits(pricingBounds.max, decimals)),
+    }
+
+    const selectedTiers = getSelectedCountryTiersFormAudienceInput(location)
+
+    const selectedTiersOrdered = Object.entries(countryTiersCoefficients)
+        .filter(([key, value]) => !!selectedTiers[key])
+        .sort((a, b) => a[1] - b[1])
+
+    // The lowest tier is with coefficient is 1 as it is already calculated in pricingBounds
+    // then we normalize the higher tiers coefficients to the lowest one
+    const normalizedCountryTiersCoefficients = Object.fromEntries(selectedTiersOrdered
+        .map(([key, value], index, entries) => {
+            if (index === 0) {
+                return [key, 1]
+            } else {
+                const normalizedCoefficient = countryTiersCoefficients[key] / countryTiersCoefficients[entries[0][0]]
+
+                return [key, normalizedCoefficient]
+            }
+
+        }))
+
+    const topSelectedTier = selectedTiersOrdered.pop()[0]
+
+
+    const rules = []
+
+    // Add price rules for each tier
+    Object.values(selectedTiers).forEach(tier => {
+        const multiplier = (normalizedCountryTiersCoefficients[tier.ruleValue])
+
+        const isTopSelectedTier = topSelectedTier === tier.ruleValue
+
+        if (multiplier !== 1 || isTopSelectedTier) {
+            const price = isTopSelectedTier ? userPricingBounds.max : userPricingBounds.min * multiplier
+            const tierPrice = getClampedNumber(price, userPricingBounds.min, userPricingBounds.max)
```

Comment:
how about if it's `== min`, you don't add the rule?

Reply:
min always has multiplier === 1, so there is no rule for min, but I'll  change it because it is not readable